### PR TITLE
Issue #11536: Added support for LITERAL_RETURN token in WhitespaceAfter

### LINKED ISSUE #11536: WhitespaceAfter should support LITERAL_RETURN token
https://checkstyle.org/config_whitespace.html#WhitespaceAfter

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac ReturnSpace.java
/var/tmp $

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAfter">
            <property name="tokens" value="LITERAL_RETURN"/>
        </module>
    </module>
</module>

/var/tmp $ cat ReturnSpace.java
public class ReturnSpace {
    int square(int x) {
        return(x * x);
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml ReturnSpace.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "LITERAL_RETURN" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "LITERAL_RETURN" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.
```

**Is your feature request related to a problem? Please describe.**
It would be nice if checkstyle could detect missing space after `return` keyword.

**Describe the solution you'd like**
WhitespaceAfter could support new literal type - `LITERAL_RETURN`

**Additional context**
#11489, #11493 and #11494 are similar requests for `try`/`finally`/`synchronized` keywords support

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Resolves #11536 
Diff Regression config: https://gist.githubusercontent.com/Rahulkhinchi03/27b23af58be50c8ac5be9e3d44db6742/raw/288d528568ed0f9959360736b0abb63ff9df14a8/config.xml
Diff Regression patch config: https://gist.githubusercontent.com/Rahulkhinchi03/f6f4d6939dd8ac952b9fce093d33ea60/raw/47b475a36fadd791c6c3fe49972771829b7feaea/patch_config.xml

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
GitHub, generate report

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
+
+public class InputWhitespaceAfterLiteralReturn {
+    public int testOne (int x){
+        return (x * x); // OK
+    }
+    public int testTwo (int x){
+        return(x * x); // violation ''return' is not followed by whitespace'
+    }
```
Comment (strkkk) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java:16]: Please add more cases like
```
boolean m1() {
        return!true;
    }

    int m2() {
        return-1;
    }

    double m3() {
        return~1;
    }

    void m4() {
        return;
    }
```

File: src/main/resources/google_checks.xml [hunk 2]
```diff
@@ -124,7 +124,8 @@
                     NOT_EQUAL, PLUS, PLUS_ASSIGN, QUESTION, RCURLY, SL, SLIST, SL_ASSIGN, SR,
                     SR_ASSIGN, STAR, STAR_ASSIGN, LITERAL_ASSERT, TYPE_EXTENSION_AND"/>
       <message key="ws.notFollowed"
-              value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only be represented as '{}' when not part of a multi-block statement (4.1.3)"/>
+              value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only
+               be represented as '{}' when not part of a multi-block statement (4.1.3)"/>
```
Comment (strkkk) [inline src/main/resources/google_checks.xml:128]: how this is related to issue?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
+
+public class InputWhitespaceAfterLiteralReturn {
+    public int testOne (int x){
+        return (x * x); // OK
+    }
+    public int testTwo (int x){
```
Comment (strkkk) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java:14]: please use correct spacing for method signature. Same for method above
`testTwo(int x) {`

**Review by strkkk — CHANGES_REQUESTED** [review]

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java (hunk 3 above)
Comment (Rahulkhinchi03) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java]: Changes Done

File: src/main/resources/google_checks.xml (hunk 2 above)
Comment (Rahulkhinchi03) [reply src/main/resources/google_checks.xml]: @strkkk @romani It is not related to this issue but this is giving me an error, So I created a new commit for this formatting.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java [hunk 4]
```diff
@@ -110,6 +112,9 @@
  *
  *      synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
  *      synchronized (this) { } // ok
+ *
+ *      return (x * x); // OK
+ *      return(x * x); // violation ''return' is not followed by whitespace'
```
Comment (nick-mancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:117]: Please do not use non-compilable code in examples. You can use a couple of simple lambdas here to show check behavior for return statements.

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Changes Done

Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Not done. A void method in Java can not have a return statement that returns a value, and no method in Java can have a method with two returns. The second return statement is always unreachable, and will not compile.

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Please check now.

Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Still not done. See https://github.com/checkstyle/checkstyle/pull/11538#discussion_r847958594 .

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java (hunk 1 above)
Comment (strkkk) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java]: not done, I do not see case with `return;`

**Comment by strkkk** [comment]
@Rahulkhinchi03 there is no regression report, please provide it.

**Comment by Rahulkhinchi03** [comment]
GitHub, generate report

**Comment by Rahulkhinchi03** [comment]
GitHub, generate report

**Comment by strkkk** [comment]
Please fix CI issue

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 4 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: @Rahulkhinchi03 I see that you changed example. Please make it with parens as before. According to report, it is most common case.

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: @strkkk I am using `return(x * x)` as an example. Do you want me to use an example with Lambda? If yes, I was getting error of bad use.

Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:117]: You do not have to use a lambda, but having unreachable/ non-compilable code in an example is not good. 

You could also add two very simple methods to this example, one ok and one with violation.

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: @Rahulkhinchi03 please change 2 consecutive returns to
```
String m1() {
   return ("a" + "b");
}

String m2() {
   return("a" + "b");
}
```

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Okay I am doing the changes but please guide me: What is the problem with my example?

**Comment by Rahulkhinchi03** [comment]
GitHub, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 4 above)
Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: @Rahulkhinchi03 if you ever have doubts, create a class with code example like this and try to compile it. javac will tell you where your errors are.

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Changes Done

Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: > create a class with code example like this and try to compile it

Please do this.

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: @nick-mancuso I have done that. It says that the result is ignored. Thank you for making me understand!

File: src/main/resources/google_checks.xml [hunk 5]
```diff
@@ -104,9 +104,9 @@
     </module>
     <module name="WhitespaceAfter">
       <property name="tokens"
-               value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE,
-                    LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, LITERAL_FINALLY, DO_WHILE, ELLIPSIS,
-                    LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_CATCH, LAMBDA"/>
+               value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE, LITERAL_WHILE, LITERAL_DO,
```
Comment (romani) [inline src/main/resources/google_checks.xml:107]: Revert this change

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/resources/google_checks.xml (hunk 5 above)
Comment (Rahulkhinchi03) [reply src/main/resources/google_checks.xml]: Changes Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 4 above)
Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: @Rahulkhinchi03 please do https://github.com/checkstyle/checkstyle/pull/11538#discussion_r849756475

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Please check. It is already done.

Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Make sure that you have pushed your changes, non compilable code is still in documentation.

**Review by romani — APPROVED** [review]
Ok to merge

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 4 above)
Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: I had a different example in the documentation before then I changed my code to https://github.com/checkstyle/checkstyle/pull/11538#discussion_r849756475 after the review.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    double testFive() {
+        return~1; // violation ''return' is not followed by whitespace'
+    }
+
+    void testSix() {
+        return; // OK
+    }
+}
```
Comment (pbludov) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java:35]: Please add
```java
    void testSix() {
        return ; // OK
    }
```
This should be a valid case for this check.

Comment (Rahulkhinchi03) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralReturn.java:35]: Changes Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 4 above)
Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: I can see that you have changed the documentation, but you did not use the examples given to you at https://github.com/checkstyle/checkstyle/pull/11538#discussion_r849756475 and comments at https://github.com/checkstyle/checkstyle/pull/11538#discussion_r847958594 still apply.

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: I am using the return in void method but not returning a value. For documentation, you mean to  say that instead of just using   
```
return("a" + "b");  // violation ''return' is not followed by whitespace'
return ("a" + "b");  // ok
```

I should use this in documentation:
```
    String testOne() { return ("a" + "b");  } // ok
    String testOne() { return("a" + "b");  } // violation ''return' is not followed by whitespace'
```

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java [hunk 7]
```diff
@@ -110,6 +112,9 @@
  *
  *      synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
  *      synchronized (this) { } // ok
+ *
+ *      String testOne() { return ("a" + "b");} // OK
```
Comment (nick-mancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:116]: Java does not allow directly nested methods. This will still not compile.

Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:116]: You are writing a method inside of another method. The two methods you added need to be declared outside of the last curly brace of the existing method.

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:116]:
```
 *      synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
 *      synchronized (this) { } // ok
 *
 *  }
 *  public String testOne() { return ("a" + "b");} // OK
 *  public String testTwo() {
 *  return("a" + "b"); // violation 'return' is not followed by whitespace'
 *  }
```

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:116]: Should I change the example? Please guide me on what I can do!

**Review by pbludov — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 7 above)
Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Please do https://github.com/checkstyle/checkstyle/pull/11538#discussion_r851731751, we do not need a separate example for this.

Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Changes Done

**Review by nick-mancuso — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 7 above)
Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Resolved.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 4 above)
Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Resolved.
